base/file.jl

```diff
@@ -578,21 +578,40 @@ end
 
 else # !windows
 
+# Based of musl __randname
+function _rand_string()
+    r = Base.rand(UInt) # FIXME: https://git.musl-libc.org/cgit/musl/tree/src/temp/__randname.c
```

Comment:
Any good ideas on how to get a random number here? Musl use `time_ns`, but then mixes in two pointers as sources of randomness.

Reply:
Is there a particular reason we are avoiding `Base.rand` ?

Reply:
Even if we use `Base.rand`, we might want to initialize our own seed since we don't want the caller to be able to override this, no?

Reply:
>  since we don't want the caller to be able to override this, no?

I don't think we care about the state of the rng. IIUC we just want a random seed that is not likely to be used by another process. This is why musl mixes in the two pointers to get some randomness in the higher bits, but we don't need to high quality randomness. 

> Is there a particular reason we are avoiding `Base.rand` ?

`Base.rand` is implemented through the stdlib `Random` and thus we would add a hard-dependency edge. Otherwise I would just use `Random.randstring`.

Reply:
I believe `Libc.rand()` should provide suitable quality for this. It appears we can also use the same code everywhere now, rather than using separate, identical implementation for windows.